api/v1alpha1/workstatus_types.go

```diff
@@ -31,13 +31,30 @@ type WorkStatus struct {
 
 	Spec   WorkStatusSpec `json:"spec,omitempty"`
 	Status RawStatus      `json:"status,omitempty"`
+	StatusDetails StatusDetails `json:"statusDetails,omitempty"`
 }
 
 // Workstatus spec
 type WorkStatusSpec struct {
 	SourceRef SourceRef `json:"sourceRef,omitempty"`
 }
 
+// StatusDetails contains information about the last applied downsync propagation.
```

Comment:
This comment is a little too specific. The `LastGeneration` field is about something that is not necessarily applied.